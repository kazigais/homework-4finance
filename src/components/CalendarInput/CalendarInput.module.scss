.input {
  width: 100%;
  padding-bottom: 10px;
}

.fluid {
  width: 100%;
}

.calendar-container {
  width: 100%;
  overflow: hidden;
}