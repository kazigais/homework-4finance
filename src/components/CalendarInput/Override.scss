.react-datepicker__month-container {
  width: 100%;
}