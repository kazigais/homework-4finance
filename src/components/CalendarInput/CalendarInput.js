import React, { PureComponent } from 'react';
import DatePicker from 'react-datepicker';
import { FaCalendarAlt } from 'react-icons/fa';
import { Input, Label } from 'semantic-ui-react';
import moment from 'moment';
import './Override.scss';
import Fade from 'react-reveal/Fade';
import PropTypes from 'prop-types';
import styles from './CalendarInput.module.scss';

class CalendarInput extends PureComponent {
  static propTypes = {
    onChange: PropTypes.func,
    value: PropTypes.instanceOf(Date),
  }

  static defaultProps = {
    onChange: () => {},
    value: new Date(),
  }

  constructor(props) {
    super(props);
    this.state = {
      maxDate: moment(props.value).add(30, 'days').toDate(),
      showDatePicker: false,
    };
  }

  toggleCalendar = () => {
    this.setState((prevState) => ({ showDatePicker: !prevState.showDatePicker }));
  }

  setStartDate = (date) => {
    this.setState({ showDatePicker: false });
    this.props.onChange(date);
  }

  render() {
    const { value } = this.props;
    return (
      <>
        <Input
          className={styles.input}
          onClick={() => this.toggleCalendar()}
          value={moment(value).format('LL')}
          labelPosition="left"
          type="text"
          placeholder="Loan length"
        >
          <Label basic><FaCalendarAlt /></Label>
          <input />
        </Input>
        <div className={styles['calendar-container']}>
          <Fade collapse top when={this.state.showDatePicker}>
            <DatePicker
              selected={value}
              maxDate={this.state.maxDate}
              minDate={value}
              onChange={(date) => this.setStartDate(date)}
              calendarClassName={styles.fluid}
              inline
            />
          </Fade>
        </div>
      </>
    );
  }
}

export default CalendarInput;
