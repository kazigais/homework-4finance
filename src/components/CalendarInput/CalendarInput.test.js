import React from 'react';
import CalendarInput from './CalendarInput';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('CalendarInput component', () => {

  let props;

  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      value: new Date('Feb 28 2013 19:00:00 EST'),
    };
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<CalendarInput {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('closes when setStartDate is called', () => {
    const wrapper = shallow(<CalendarInput {...props} />);
    wrapper.state().showDatePicker = true;
    wrapper.instance().setStartDate(props.value);
    expect(wrapper.state().showDatePicker).toBe(false);
  });

  it('calls handler when setStartDate is called', () => {
    const wrapper = shallow(<CalendarInput {...props} />);
    wrapper.instance().setStartDate(props.value);
    expect(props.onChange).toHaveBeenCalled();
  });

  it('toggles modal state when toggleCalendar is called', () => {
    const wrapper = shallow(<CalendarInput {...props} />);
    wrapper.instance().toggleCalendar();
    expect(wrapper.state().showDatePicker).toBe(true);
  });
});
