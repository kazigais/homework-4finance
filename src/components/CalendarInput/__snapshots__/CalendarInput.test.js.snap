// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarInput component matches snapshot 1`] = `
<Fragment>
  <Input
    className="input"
    labelPosition="left"
    onClick={[Function]}
    placeholder="Loan length"
    type="text"
    value="February 28, 2013"
  >
    <Label
      basic={true}
    >
      <FaCalendarAlt />
    </Label>
    <input />
  </Input>
  <div
    className="calendar-container"
  >
    <Fade
      collapse={true}
      top={true}
      when={false}
    >
      <t
        allowSameDay={false}
        calendarClassName="fluid"
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inline={true}
        inlineFocusSelectedMonth={false}
        maxDate={2013-03-30T23:00:00.000Z}
        minDate={2013-03-01T00:00:00.000Z}
        monthsShown={1}
        nextMonthButtonLabel="Next Month"
        nextYearButtonLabel="Next Year"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        previousYearButtonLabel="Previous Year"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2013-03-01T00:00:00.000Z}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showQuarterYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </Fade>
  </div>
</Fragment>
`;
